tiledb/sm/query/query_condition.cc

```diff
@@ -1564,6 +1611,12 @@ void QueryCondition::apply_ast_node_sparse(
   const void* condition_value_content =
       node->get_condition_value_view().content();
   const size_t condition_value_size = node->get_condition_value_view().size();
+  uint8_t* buffer_validity = nullptr;
+  if constexpr (
```

Comment:
Everywhere you have std::is_same_v<CombinationOp, QCMax<BitmapType>>, could you add a comment saying that this is for OR CombinationOp? That way someone new that is looking at this code will not have to try to figure it out.